File: articles/neural-networks-and-representation-learning/calculators/r/representation_score_calculator.R
args <- commandArgs(trailingOnly = TRUE)
values <- as.numeric(args)
if (length(values) < 4) {
  stop("Usage: Rscript representation_score_calculator.R x1 x2 x3 bias")
}
x1 <- values[1]
x2 <- values[2]
x3 <- values[3]
bias <- values[4]
linear <- 0.9 * x1 - 0.7 * x2 + 0.35 * x3 + bias
score <- 1 / (1 + exp(-linear))
cat(sprintf("linear_signal=%.6f\n", linear))
cat(sprintf("representation_score=%.6f\n", score))

File: articles/neural-networks-and-representation-learning/calculators/python/representation_score_calculator.py
from __future__ import annotations
import argparse
import math


def sigmoid(value: float) -> float:
    return 1.0 / (1.0 + math.exp(-value))


def main() -> None:
    parser = argparse.ArgumentParser(description="Compute a simple neural-style representation score.")
    parser.add_argument("--x1", type=float, required=True)
    parser.add_argument("--x2", type=float, required=True)
    parser.add_argument("--x3", type=float, required=True)
    parser.add_argument("--w1", type=float, default=0.9)
    parser.add_argument("--w2", type=float, default=-0.7)
    parser.add_argument("--w3", type=float, default=0.35)
    parser.add_argument("--bias", type=float, default=0.0)
    args = parser.parse_args()
    linear = args.w1 * args.x1 + args.w2 * args.x2 + args.w3 * args.x3 + args.bias
    score = sigmoid(linear)
    print(f"linear_signal={linear:.6f}")
    print(f"representation_score={score:.6f}")


if __name__ == "__main__":
    main()
